From 741509b3dd7f68e3dad58043644ca4736cc90e27 Mon Sep 17 00:00:00 2001
From: "R, Kannappan" <r.kannappan@intel.com>
Date: Fri, 10 Oct 2014 10:47:25 +0530
Subject: [PATCH 1/1] power-supply-charger: fix null pointer exception

in otg notification input data pointer could be null.
validate the input before using.

Change-Id: I77666d6b93b933ac23e718ac30db34a7d63e78bf
Signed-off-by: R, Kannappan <r.kannappan@intel.com>
---
 drivers/power/power_supply_charger.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/power/power_supply_charger.c b/drivers/power/power_supply_charger.c
index 23515c0ef05d..bd21e62fd8a1 100644
--- a/drivers/power/power_supply_charger.c
+++ b/drivers/power/power_supply_charger.c
@@ -162,11 +162,11 @@ static int handle_cable_notification(struct notifier_block *nb,
 {
 	struct power_supply_cable_props cap;
 
-	memcpy(&cap, data, sizeof(struct power_supply_cable_props));
-
 	if (event != USB_EVENT_CHARGER && event != PSY_CABLE_EVENT)
 		return NOTIFY_DONE;
 
+	if (data)
+		memcpy(&cap, data, sizeof(struct power_supply_cable_props));
 	process_cable_props(&cap);
 
 	return NOTIFY_OK;
-- 
1.7.9.5

